packages/btc/src/utils.ts

```diff
@@ -11,14 +15,14 @@ function tapTweakHash(publicKey: Buffer, hash: Buffer | undefined): Buffer {
   return bitcoin.crypto.taggedHash('TapTweak', Buffer.concat(hash ? [publicKey, hash] : [publicKey]));
 }
 
-export function tweakSigner<T extends bitcoin.Signer>(
+export function tweakSigner<T extends TweakableSigner>(
   signer: T,
   options?: {
     network?: bitcoin.Network;
     tweakHash?: Buffer;
   },
 ): bitcoin.Signer {
-  let privateKey: Uint8Array | undefined = (signer as any).privateKey;
+  let privateKey: Uint8Array | undefined = signer.privateKey ? new Uint8Array(signer.privateKey) : undefined;
   if (!privateKey) {
     throw new Error('Private key is required for tweaking signer!');
```

Comment:
By exchanging these two lines, we don't need to check whether it is undefined in `let privateKey`, and the type can also be determined to be Uint8Array.